dsgrid/config/dataset_schema_handler_base.py

```diff
@@ -31,6 +91,34 @@ def get_pivot_dimension_columns(self):
         dim_type = self._config.model.data_schema.load_data_column_dimension
         return sorted(list(self._config.get_dimension(dim_type).get_unique_ids()))
 
+    def get_pivot_dimension_columns_mapped_to_project(self, mapping_references):
+        """Get cols for the dimension that is pivoted in load_data.
+
+        Returns
+        -------
+        dict
+            Mapping of pivoted dimension column names to project record names.
+
+        """
+        columns = set(self.get_pivot_dimension_columns())
+        dim_type = self.get_pivot_dimension_type()
+        for ref in mapping_references:
+            if ref.from_dimension_type.value == dim_type:
+                mapping_config = self._dimension_mapping_mgr.get_by_id(
+                    ref.mapping_id, version=ref.version
+                )
+                mapping = {
+                    x.from_id: x.to_id for x in mapping_config.model.records if x.to_id is not None
```

Comment:
If `to_id` is None then the record cannot be present in the pivoted columns. The dict returned from this method will only include actual columns. This is just for the check on line 114. Let me know if I'm missing something.

I do think there is a bug here. `to_id` may be `""` instead of `None` because the `AssociationTableRecordModel` defines the field as a string. I think we need to coerce these values to `None`.

Reply:
I meant to say that `diff = columns.symmetric_difference(mapping)` should be done before anything is removed from the mapping. Also, shouldn't columns be compared against the `from_id` only in the mapping?